Move to PSR-4 autoload

barryvdh opened the pull request:
PSR-0 is deprecated and PSR-4 is easier :)
https://philsturgeon.uk/php/2014/07/19/send-psr0-to-the-standards-farm-in-the-sky/

bsweeney commented:
Can you rebase your changes to the latest dev (sorry about that). I'll merge once that's done.

barryvdh commented:
I've rebased and also changed the autoloader (forgot that before).

But please do check this is correct, in case I missed any references or changes with rebasing.

bsweeney commented:
I'll give it a look, but the changes were fairly minor so I don't expect any conflicts.
